Maven 3.10.0-rc1 compatibility

Flatten maven plugin fails with maven 3.10.0-rc1:

```
[ERROR] Failed to execute goal org.codehaus.mojo:flatten-maven-plugin:1.7.3:flatten (flatten) on project parent: failed to create a clean pom: unable to create flattened dependencies: caught exception when flattening dependencies: Failed to read artifact descriptor for org.junit.jupiter:junit-jupiter::6.1.0: The following artifacts could not be resolved: org.junit.jupiter:junit-jupiter:pom:6.1.0 (present, but unavailable) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:flatten-maven-plugin:1.7.3:flatten (flatten) on project parent: failed to create a clean pom
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:342)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:324)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:220)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:181)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:79)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:169)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:166)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:106)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:74)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:121)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:179)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:1017)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:299)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:219)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoExecutionException: failed to create a clean pom
    at org.codehaus.mojo.flatten.FlattenMojo.createCleanPom (FlattenMojo.java:656)
    at org.codehaus.mojo.flatten.FlattenMojo.access$700 (FlattenMojo.java:212)
    at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getCleanPom (FlattenMojo.java:1387)
    at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getModel (FlattenMojo.java:1405)
    at org.codehaus.mojo.flatten.FlattenMojo.getSourceModel (FlattenMojo.java:799)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedPom (FlattenMojo.java:629)
    at org.codehaus.mojo.flatten.FlattenMojo.execute (FlattenMojo.java:485)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:127)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:336)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:324)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:220)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:181)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:79)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:169)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:166)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:106)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:74)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:121)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:179)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:1017)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:299)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:219)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoExecutionException: unable to create flattened dependencies
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1036)
    at org.codehaus.mojo.flatten.FlattenMojo.createCleanPomImpl (FlattenMojo.java:789)
    at org.codehaus.mojo.flatten.FlattenMojo.createCleanPom (FlattenMojo.java:654)
    at org.codehaus.mojo.flatten.FlattenMojo.access$700 (FlattenMojo.java:212)
    at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getCleanPom (FlattenMojo.java:1387)
    at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getModel (FlattenMojo.java:1405)
    at org.codehaus.mojo.flatten.FlattenMojo.getSourceModel (FlattenMojo.java:799)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedPom (FlattenMojo.java:629)
    at org.codehaus.mojo.flatten.FlattenMojo.execute (FlattenMojo.java:485)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:127)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:336)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:324)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:220)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:181)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:79)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:169)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:166)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:106)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:74)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:121)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:179)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:1017)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:299)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:219)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoExecutionException: caught exception when flattening dependencies
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1283)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1034)
    at org.codehaus.mojo.flatten.FlattenMojo.createCleanPomImpl (FlattenMojo.java:789)
    at org.codehaus.mojo.flatten.FlattenMojo.createCleanPom (FlattenMojo.java:654)
    at org.codehaus.mojo.flatten.FlattenMojo.access$700 (FlattenMojo.java:212)
    at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getCleanPom (FlattenMojo.java:1387)
    at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getModel (FlattenMojo.java:1405)
    at org.codehaus.mojo.flatten.FlattenMojo.getSourceModel (FlattenMojo.java:799)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedPom (FlattenMojo.java:629)
    at org.codehaus.mojo.flatten.FlattenMojo.execute (FlattenMojo.java:485)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:127)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:336)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:324)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:220)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:181)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:79)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:169)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:166)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:106)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:74)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:121)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:179)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:1017)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:299)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:219)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for org.junit.jupiter:junit-jupiter::6.1.0
    at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom (DefaultArtifactDescriptorReader.java:338)
    at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor (DefaultArtifactDescriptorReader.java:263)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.readArtifactDescriptor (DefaultRepositorySystem.java:206)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependenciesAll (FlattenMojo.java:1194)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1276)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1034)
    at org.codehaus.mojo.flatten.FlattenMojo.createCleanPomImpl (FlattenMojo.java:789)
    at org.codehaus.mojo.flatten.FlattenMojo.createCleanPom (FlattenMojo.java:654)
    at org.codehaus.mojo.flatten.FlattenMojo.access$700 (FlattenMojo.java:212)
    at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getCleanPom (FlattenMojo.java:1387)
    at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getModel (FlattenMojo.java:1405)
    at org.codehaus.mojo.flatten.FlattenMojo.getSourceModel (FlattenMojo.java:799)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedPom (FlattenMojo.java:629)
    at org.codehaus.mojo.flatten.FlattenMojo.execute (FlattenMojo.java:485)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:127)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:336)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:324)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:220)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:181)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:79)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:169)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:166)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:106)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:74)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:121)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:179)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:1017)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:299)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:219)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    Suppressed: org.eclipse.aether.resolution.ArtifactResolutionException: The following artifacts could not be resolved: org.junit.jupiter:junit-jupiter:pom:6.1.0 (present, but unavailable)
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:425)
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:201)
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:180)
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom (DefaultArtifactDescriptorReader.java:327)
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor (DefaultArtifactDescriptorReader.java:263)
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.readArtifactDescriptor (DefaultRepositorySystem.java:206)
        at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependenciesAll (FlattenMojo.java:1194)
        at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1276)
        at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1034)
        at org.codehaus.mojo.flatten.FlattenMojo.createCleanPomImpl (FlattenMojo.java:789)
        at org.codehaus.mojo.flatten.FlattenMojo.createCleanPom (FlattenMojo.java:654)
        at org.codehaus.mojo.flatten.FlattenMojo.access$700 (FlattenMojo.java:212)
        at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getCleanPom (FlattenMojo.java:1387)
        at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getModel (FlattenMojo.java:1405)
        at org.codehaus.mojo.flatten.FlattenMojo.getSourceModel (FlattenMojo.java:799)
        at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedPom (FlattenMojo.java:629)
        at org.codehaus.mojo.flatten.FlattenMojo.execute (FlattenMojo.java:485)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:127)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:336)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:324)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:220)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:181)
        at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:79)
        at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:169)
        at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:166)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:106)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:74)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:121)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:179)
        at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute (MavenCli.java:1017)
        at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:299)
        at org.apache.maven.cli.MavenCli.main (MavenCli.java:219)
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
        at java.lang.reflect.Method.invoke (Method.java:565)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
        Suppressed: org.eclipse.aether.resolution.ArtifactResolutionException: Failed to resolve artifact org.junit.jupiter:junit-jupiter:pom:6.1.0
            at org.eclipse.aether.resolution.ArtifactResolutionException.getSuppressed (ArtifactResolutionException.java:185)
            at org.eclipse.aether.resolution.ArtifactResolutionException.<init> (ArtifactResolutionException.java:47)
            at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:425)
            at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:201)
            at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:180)
            at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom (DefaultArtifactDescriptorReader.java:327)
            at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor (DefaultArtifactDescriptorReader.java:263)
            at org.eclipse.aether.internal.impl.DefaultRepositorySystem.readArtifactDescriptor (DefaultRepositorySystem.java:206)
            at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependenciesAll (FlattenMojo.java:1194)
            at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1276)
            at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1034)
            at org.codehaus.mojo.flatten.FlattenMojo.createCleanPomImpl (FlattenMojo.java:789)
            at org.codehaus.mojo.flatten.FlattenMojo.createCleanPom (FlattenMojo.java:654)
            at org.codehaus.mojo.flatten.FlattenMojo.access$700 (FlattenMojo.java:212)
            at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getCleanPom (FlattenMojo.java:1387)
            at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getModel (FlattenMojo.java:1405)
            at org.codehaus.mojo.flatten.FlattenMojo.getSourceModel (FlattenMojo.java:799)
            at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedPom (FlattenMojo.java:629)
            at org.codehaus.mojo.flatten.FlattenMojo.execute (FlattenMojo.java:485)
            at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:127)
            at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:336)
            at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:324)
            at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:220)
            at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:181)
            at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:79)
            at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:169)
            at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
            at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:166)
            at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:106)
            at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:74)
            at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
            at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:121)
            at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
            at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:179)
            at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
            at org.apache.maven.cli.MavenCli.execute (MavenCli.java:1017)
            at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:299)
            at org.apache.maven.cli.MavenCli.main (MavenCli.java:219)
            at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
            at java.lang.reflect.Method.invoke (Method.java:565)
            at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
            at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
            at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
            at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: The following artifacts could not be resolved: org.junit.jupiter:junit-jupiter:pom:6.1.0 (present, but unavailable)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:425)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:201)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:180)
    at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom (DefaultArtifactDescriptorReader.java:327)
    at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor (DefaultArtifactDescriptorReader.java:263)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.readArtifactDescriptor (DefaultRepositorySystem.java:206)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependenciesAll (FlattenMojo.java:1194)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1276)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1034)
    at org.codehaus.mojo.flatten.FlattenMojo.createCleanPomImpl (FlattenMojo.java:789)
    at org.codehaus.mojo.flatten.FlattenMojo.createCleanPom (FlattenMojo.java:654)
    at org.codehaus.mojo.flatten.FlattenMojo.access$700 (FlattenMojo.java:212)
    at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getCleanPom (FlattenMojo.java:1387)
    at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getModel (FlattenMojo.java:1405)
    at org.codehaus.mojo.flatten.FlattenMojo.getSourceModel (FlattenMojo.java:799)
    at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedPom (FlattenMojo.java:629)
    at org.codehaus.mojo.flatten.FlattenMojo.execute (FlattenMojo.java:485)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:127)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:336)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:324)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:220)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:181)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:79)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:169)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:166)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:106)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:74)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:121)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:179)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:1017)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:299)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:219)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
    Suppressed: org.eclipse.aether.resolution.ArtifactResolutionException: Failed to resolve artifact org.junit.jupiter:junit-jupiter:pom:6.1.0
        at org.eclipse.aether.resolution.ArtifactResolutionException.getSuppressed (ArtifactResolutionException.java:185)
        at org.eclipse.aether.resolution.ArtifactResolutionException.<init> (ArtifactResolutionException.java:47)
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:425)
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:201)
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:180)
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom (DefaultArtifactDescriptorReader.java:327)
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor (DefaultArtifactDescriptorReader.java:263)
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.readArtifactDescriptor (DefaultRepositorySystem.java:206)
        at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependenciesAll (FlattenMojo.java:1194)
        at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1276)
        at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedDependencies (FlattenMojo.java:1034)
        at org.codehaus.mojo.flatten.FlattenMojo.createCleanPomImpl (FlattenMojo.java:789)
        at org.codehaus.mojo.flatten.FlattenMojo.createCleanPom (FlattenMojo.java:654)
        at org.codehaus.mojo.flatten.FlattenMojo.access$700 (FlattenMojo.java:212)
        at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getCleanPom (FlattenMojo.java:1387)
        at org.codehaus.mojo.flatten.FlattenMojo$ModelsFactory.getModel (FlattenMojo.java:1405)
        at org.codehaus.mojo.flatten.FlattenMojo.getSourceModel (FlattenMojo.java:799)
        at org.codehaus.mojo.flatten.FlattenMojo.createFlattenedPom (FlattenMojo.java:629)
        at org.codehaus.mojo.flatten.FlattenMojo.execute (FlattenMojo.java:485)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:127)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:336)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:324)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:220)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:181)
        at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:79)
        at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:169)
        at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:166)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:106)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:74)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:121)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:267)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:179)
        at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute (MavenCli.java:1017)
        at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:299)
        at org.apache.maven.cli.MavenCli.main (MavenCli.java:219)
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
        at java.lang.reflect.Method.invoke (Method.java:565)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:362)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```

--- a/src/it/projects/flatten-deploy/invoker.properties
+++ b/src/it/projects/flatten-deploy/invoker.properties
@@ -1,1 +1,4 @@
 invoker.goals=clean deploy
+
+# FIXME Failed with Maven 4.0.0-rc-5
+invoker.maven.version = !4.0.0+

--- a/src/it/projects/resolve-properties-ci/invoker.properties
+++ b/src/it/projects/resolve-properties-ci/invoker.properties
@@ -1,3 +1,3 @@
 invoker.mavenOpts = -Drevision=1.2.3.4 -Drevision1=1.2.3.5 -Dsmall=small -Dsha1=abcdefg -Dchangelist=1,2,3,4
-# Maven 4 requires all properties to be resolved in dependencies
-invoker.maven.version = !4.0.0+
+# Maven 4 and 3.10.0 requires all properties to be resolved in dependencies
+invoker.maven.version = !4.0.0+,!3.10.0

--- a/src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java
+++ b/src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java
@@ -1198,7 +1198,8 @@ public class FlattenMojo extends AbstractFlattenMojo {
 
                 org.eclipse.aether.artifact.Artifact aetherArtifact = new DefaultArtifact(
                         artifact.getGroupId(), artifact.getArtifactId(), null, artifact.getVersion());
-                ArtifactDescriptorRequest request = new ArtifactDescriptorRequest(aetherArtifact, null, null);
+                ArtifactDescriptorRequest request =
+                        new ArtifactDescriptorRequest(aetherArtifact, project.getRemoteProjectRepositories(), null);
                 ArtifactDescriptorResult artifactDescriptorResult =
                         repositorySystem.readArtifactDescriptor(this.session.getRepositorySession(), request);
 

